### ISSUE TITLE
Add support for lazy-loading of NPM modules to loadChildren

### ISSUE THREAD

**Issue by adrianbarwicki** [angular/angular#38340]
# 🚀 feature request

### Relevant Package

This feature request is for `@angular/router@9.1.12`.

### Description

Lazy loading is possible with the following use of the `RouterModule`:
```ts
        RouterModule.forChild([
            {
                path: "lazy",
                loadChildren: () =>
                    import("./lazy.module").then(
                        (m) => m.LazyModule
                    ),
            },
        ]),
```

We build the `lazy.module.ts` and publish it as an npm package with `ng-packagr`. Then we import the npm package in the following way for lazy loading:
```ts
        RouterModule.forChild([
            {
                path: "lazy",
                loadChildren: () =>
                    import("lazy-module").then(
                        (m) => m.LazyModule
                    ),
            },
        ]),
```
This does not work.

When navigating to the path `lazy` programmatically with the logic `this.router.navigateByUrl("lazy")` nothing happens and there are no error logs. However, it can be seen in the network tab that the module is indeed loaded.

We build the app with Ivy enabled:
- in angular.json:
```json
{
  "architect": {
      "build": {
          "configurations": {
              "envName": {
                  "aot": true,
                  "optimization": true,
                  "buildOptimizer": true
              }
          }
      }
  }
}
```
- In `tsconfig.json`, set `angularCompilerOptions.useIvy` to `true`.

### Describe the solution you'd like

We'd like the following import works:
```ts
        RouterModule.forChild([
            {
                path: "lazy",
                loadChildren: () =>
                    import("lazy-module").then( // "lazy-module" is an npm package
                        (m) => m.LazyModule
                    ),
            },
        ]),
```

### Describe alternatives you've considered
At the moment, we are using the following workaround. We wrap the npm module in an angular module and import it. This leads to a significant increase of code, as most of our modules are published to a npm registry and imported that way.
```ts
import {NgModule} from "@angular/core";
import {LazyModule} from "lazy-module";

@NgModule({
    imports: [
        LazyModule
    ],
})
export class WrapperLazyModule {}
```

**Comment by alan-agius4** [comment]
Hi @adrianbarwicki, can you please elaborate on the failure? Also which version of Angular 9-10 with Ivy? 

I don't see any reason why this should not work with Ivy.

**Comment by adrianbarwicki** [comment]
> Hi @adrianbarwicki, can you please elaborate on the failure? Also which version of Angular 9-10 with Ivy?
> 
> I don't see any reason why this should not work with Ivy.

Hi @alan-agius4, I extended the first comment.

We use `@angular/router@9.1.12`.

When navigating to the path `lazy` programmatically with the logic `this.router.navigateByUrl("lazy")` nothing happens and there are no error logs. However, it can be seen in the network tab that the module gets indeed loaded.

We do use Ivy for the builds. We set in angular.json
```
"aot": true,
"optimization": true,
"buildOptimizer": true
```
and in the `tsconfig.json` we set `angularCompilerOptions.useIvy` to `true`.
